executor/common_linux.h

```diff
@@ -1626,6 +1626,377 @@ static long syz_genetlink_get_family_id(volatile long name)
+	uint8 status;
+	uint16 handle;
+	uint8 features[8];
+} __attribute__((packed)) evt_read_remote_features_complete;
+#define EVT_READ_REMOTE_FEATURES_COMPLETE_SIZE 11
+
+#define EVT_CMD_COMPLETE 0x0E
+typedef struct {
+	uint8 ncmd;
+	uint16 opcode;
+} __attribute__((packed)) evt_cmd_complete;
+#define EVT_CMD_COMPLETE_SIZE 3
+
+#define OCF_READ_BUFFER_SIZE 0x0005
+typedef struct {
+	uint8 status;
+	uint16 acl_mtu;
+	uint8 sco_mtu;
+	uint16 acl_max_pkt;
+	uint16 sco_max_pkt;
+} __attribute__((packed)) read_buffer_size_rp;
+#define READ_BUFFER_SIZE_RP_SIZE 8
+
+#define OCF_READ_BD_ADDR 0x0009
+typedef struct {
+	uint8 status;
+	bdaddr_t bdaddr;
+} __attribute__((packed)) read_bd_addr_rp;
+#define READ_BD_ADDR_RP_SIZE 7
+
+#define OCF_WRITE_CONN_ACCEPT_TIMEOUT 0x0016
+typedef struct {
+	uint16 timeout;
+} __attribute__((packed)) write_conn_accept_timeout_cp;
+#define WRITE_CONN_ACCEPT_TIMEOUT_CP_SIZE 2
+
+#define BTPROTO_HCI 1
+
+#define SCAN_PAGE 0x02
+
+#define HCISETSCAN _IOW('H', 221, int)
+
+struct hci_dev_req {
+	uint16 dev_id;
+	uint32 dev_opt;
+};
+
+struct sockaddr_hci {
+	sa_family_t hci_family;
+	unsigned short hci_dev;
+	unsigned short hci_channel;
+};
+
+static long syz_init_net_socket(volatile long domain, volatile long type, volatile long proto);
+
+int hci_open_dev(int dev_id)
+{
+	struct sockaddr_hci a;
+	int dd, err;
+
+	/* Create HCI socket */
+	dd = syz_init_net_socket(AF_BLUETOOTH, SOCK_RAW | SOCK_CLOEXEC, BTPROTO_HCI);
+	if (dd < 0)
+		return dd;
+
+	/* Bind socket to the HCI device */
+	memset(&a, 0, sizeof(a));
+	a.hci_family = AF_BLUETOOTH;
+	a.hci_dev = dev_id;
+	if (bind(dd, (struct sockaddr*)&a, sizeof(a)) < 0)
+		goto failed;
+
+	return dd;
+
+failed:
+	err = errno;
+	close(dd);
+	errno = err;
+
+	return -1;
+}
+
+static int vhci_fd = -1;
+
+int hci_send_event_packet(int fd, uint8 evt, void* data, size_t data_len)
+{
+	struct iovec iv[3];
+
+	hci_event_hdr hdr;
+	hdr.evt = evt;
+	hdr.plen = data_len;
+
+	uint8 type = HCI_EVENT_PKT;
+
+	iv[0].iov_base = &type;
+	iv[0].iov_len = sizeof(type);
+	iv[1].iov_base = &hdr;
+	iv[1].iov_len = HCI_EVENT_HDR_SIZE;
+	iv[2].iov_base = data;
+	iv[2].iov_len = data_len;
+
+	return writev(fd, iv, sizeof(iv) / sizeof(struct iovec));
+}
+
+int hci_send_event_cmd_complete(int fd, uint16 opcode, void* data, size_t data_len)
+{
+	struct iovec iv[4];
+
+	hci_event_hdr hdr;
+	hdr.evt = EVT_CMD_COMPLETE;
+	hdr.plen = EVT_CMD_COMPLETE_SIZE + data_len;
+
+	evt_cmd_complete evt_hdr;
+	evt_hdr.ncmd = 1;
+	evt_hdr.opcode = opcode;
+
+	uint8 type = HCI_EVENT_PKT;
+
+	iv[0].iov_base = &type;
+	iv[0].iov_len = sizeof(type);
+	iv[1].iov_base = &hdr;
+	iv[1].iov_len = HCI_EVENT_HDR_SIZE;
+	iv[2].iov_base = &evt_hdr;
+	iv[2].iov_len = EVT_CMD_COMPLETE_SIZE;
+	iv[3].iov_base = data;
+	iv[3].iov_len = data_len;
+
+	return writev(fd, iv, sizeof(iv) / sizeof(struct iovec));
+}
+
+static int vhci_init = 0;
+
+// TODO: fuzz event packets.
+void process_command_pkt(int fd, char* buf, size_t buf_size)
```

Comment:
This thread will live in a parent process, not in actual test processes. This may work, but it's quite strange.
Generally we try to not have background threads b/c that's source of non-determinism and may interfere with test execution in other ways.
A background thread in the parent process also does not allow to use any randomness in replies.

Have you considered doing the same we do for USB? Namely, add a pseudo-syscall that reads 1 even and replies to synchronously, potentially using some fuzzer randomness (e.g. for the address, status, etc).